src/elfpy/simulators.py

```diff
@@ -1,26 +1,104 @@
 """Simulator class wraps the pricing models and markets for experiment tracking and execution"""
 
 from __future__ import annotations  # types will be strings by default in 3.11
-from typing import Optional, TYPE_CHECKING
+from typing import Callable, Optional, TYPE_CHECKING
 import datetime
 import json
 import logging
-
 import numpy as np
 from numpy.random._generator import Generator
+from stochastic.processes import GeometricBrownianMotion
 
-from elfpy.utils import sim_utils  # utilities for setting up a simulation
 import elfpy.utils.time as time_utils
 from elfpy.utils.outputs import CustomEncoder
-from elfpy.types import MarketDeltas, SimulationState
+from elfpy.types import MarketDeltas, RandomSimulationVariables, SimulationState
 
 if TYPE_CHECKING:
     from elfpy.agent import Agent
     from elfpy.markets import Market
-    from elfpy.pricing_models.base import PricingModel
     from elfpy.utils.config import Config
 
 
+def setup_vault_apr(config, rng):
```

Comment:
What is the reason for putting `setup_vault_apr` and `get_random_variables` here instead of in `sim_utils`?

Reply:
Circular imports.

Reply:
If that is a result of typing you can resolve circular imports by placing the import statements behind a `TYPE_CHECKING` flag, which is already done for other imports in `simulators.py`. I would prefer this for a few reasons: `simulators.py` can be limited in scope to just the simulator class; all of the `get` functions can live together as utils; and imports are more consistent between the chief class objects and helper functions.

Reply:
That being said, if the problem is not resolved by the TYPE_CHECKING flag then I'm fine with punting this to a later PR